FirebaseVertexAI/Sources/Types/Public/Imagen/ImagenGenerationConfig.swift

```diff
@@ -0,0 +1,24 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+@available(iOS 15.0, macOS 12.0, macCatalyst 15.0, tvOS 15.0, watchOS 8.0, *)
+public struct ImagenGenerationConfig {
+  public var numberOfImages: Int?
+  public var negativePrompt: String?
+  public var aspectRatio: ImagenAspectRatio?
+  public var imageFormat: ImagenImageFormat?
+  public var addWatermark: Bool?
```

Comment:
Note: Still considering whether these should be mutable. It does make it significantly easier to adjust individual parameters and the config can be assigned to a `let` if the developer wants to keep them immutable.

Reply:
Just thinking aloud here... I've been thinking about what configuration types like this should look like in Swift 6. Here is my current thinking on the options:
1. mark unchecked sendable and make each property atomic with a corresponding NSLock that guards a backing ivar (sync)
2. Make the Config structure an actor (async)
3. Mark the Config structure with a global actor (e.g. VertexAIActor) (async)
4. Make a ConfigBuilder class that's an actor (or executes on global actor) that can build a immutable `ImageGenerationConfig` (async for the builder, sync for the produced value)